app/src/composables/useCustomFetch.ts

```diff
@@ -16,6 +20,14 @@ export const useCustomFetch = createFetch({
         }
       }
       return { options }
+    },
+    onFetchError({ data, response, error }) {
+      const status = response?.status ?? 0
+      // Store the error response status
+      errorMessage.value = status
+        ? `Error ${status}: ${getFetchErrorMessage(status)}`
+        : `Network Error: ${error?.message ?? 'An unknown error occurred'}`
+      return { data, response, error }
```

Comment:
You don't need to handle siwe error here. 

Can you look at useFetch, make an exemple to see what are the error the return ?

Create a simple project and make a fetch to an endpoint

- Request an url the not exit
- Request an endpoint that return 500 error 
- Request an endpoint that return 400 error

and look at what are the different between their response : {data, error}


Here, you are just changing the error message. 

What i want is to handle the error in the SIWE.
Like

When i get the error from useFetch (no need to change the error that useFetch return)
Then if the error is 
- 500 error display to the user :An error occured try later
- network error :There is an network error
- 400 Error: You can't make this request

And in the siwe if we have the error from Metmask return: Request Rejected by the user.


When i say show to the user i'm talking about the toast.

Reply:
I think it's easier to respond to the status codes instead of the error response because 404 error produces something like this; `Unexpected token '<', "<!DOCTYPE "... is not valid JSON`

Reply:
500 produces `Internal Server Error` and 401 produces `Unauthorized`